packages/discord.js/src/structures/ModalSubmitInteraction.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const { createComponent } = require('@discordjs/builders');
+const Interaction = require('./Interaction');
+const InteractionWebhook = require('./InteractionWebhook');
+const ModalSubmitFieldsResolver = require('./ModalSubmitFieldsResolver');
+const InteractionResponses = require('./interfaces/InteractionResponses');
+
+/**
+ * @typedef {Object} ModalFieldData
+ * @property {string} value The value of the field
+ * @property {ComponentType} type The component type of the field
+ * @property {string} customId The custom id of the field
+ */
+
+/**
+ * Represents a modal interaction
+ */
+class ModalSubmitInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+    /**
+     * The custom id of the modal.
+     * @type {string}
+     */
+    this.customId = data.data.custom_id;
+
+    if ('message' in data) {
+      /**
+       * The message associated with this interaction
+       * @type {Message|APIMessage|null}
+       */
+      this.message = this.channel?.messages._add(data.message);
```

Comment:
Wouldn't this evaluate to `undefined` if the channel is not cached? (e.g. interaction came from a guild the bot is not in)
(missing `?? data.message`?)